File: 0268-missing-number/0268-missing-number.cs
public class Solution {
    public int MissingNumber(int[] nums) {
        HashSet<int> numSet = new HashSet<int>();
        for(int i = 0;i<nums.Length;i++){
            numSet.Add(nums[i]);
        }
        for(int j = 0;j<=nums.Length;j++){
            if(!numSet.Contains(j)){
                return j;
            }
        }
        return -1;
    }
}

File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        //O(n)
        int sm1 = 0, sm2 = 0, n = nums.size();
        for(int i = 0;i<nums.size();i++){
            sm1 += nums[i];
        }
        sm2 = n * (n+1)/2;
        return sm2 - sm1;
    }
};